### PR TITLE
Improve updating and use persistent download directory

### PR DESCRIPTION
**Pull request by mohe2015**
These changes (especially the update logic) are probably questionable and need a discussion. What do you think about that idea?
This would make this package kinda like a git package. An alternative would be to write an automatic script so that you could automatically push updates every day.

### CODE REVIEW

File: thunderbird-nightly-bin/PKGBUILD [hunk 1]
```diff
@@ -23,16 +23,18 @@ install="${pkgname%-bin}.install"
...
         "vendor.js")
 
-b2sums=('SKIP'
-        'SKIP'
-        '1f954c9a23842367be37c1f56b853de6a336c1b4156df42f88b14f1614d194c2f996f1cc589cdbfcea398bb9cedcd527ef3c6fb8b01e2f07c46290043e24c6b7'
+b2sums=('1f954c9a23842367be37c1f56b853de6a336c1b4156df42f88b14f1614d194c2f996f1cc589cdbfcea398bb9cedcd527ef3c6fb8b01e2f07c46290043e24c6b7'
         '674d1ee883e675c37b0af0ac97c339a8c0f2b53cb06e64db64aaa3f22f83d7179b6fa3e122344f3413ccb9956776288db9bc608b5cedef640cbd223838be7476')
 
+prepare() {
+  /usr/bin/curl -z "${FX_SRC}.txt" -gqb "" -fL --retry 3 --retry-delay 3 -o "${FX_SRC}.txt" "${FX_SRC_URI}.txt"
```
Comment (freswa) [inline thunderbird-nightly-bin/PKGBUILD]: We don't download outside of the source array.

File: thunderbird-nightly-bin/PKGBUILD [hunk 2]
```diff
@@ -41,13 +41,13 @@ pkgver(){
 
 package() {
   install -d "${pkgdir}"/{usr/bin,opt}
-  cp -a thunderbird "${pkgdir}"/opt/${pkgname%-bin}-${pkgver}
-  cp vendor.js "${pkgdir}"/opt/${pkgname%-bin}-${pkgver}/defaults/pref/
-  ln -s /opt/${pkgname%-bin}-${pkgver}/thunderbird "${pkgdir}"/usr/bin/${pkgname%-bin}
+  cp -a thunderbird "${pkgdir}"/opt/${pkgname%-bin}
```
Comment (freswa) [inline thunderbird-nightly-bin/PKGBUILD:55]: This doesn't work with a clean chroot build....

**Comment by freswa** [comment]
TBH I don't think this has a chance, since it violates the arch packaging guidlines...

**Comment by mohe2015** [comment]
> TBH I don't think this has a chance, since it violates the arch packaging guidlines...

I more or less expected that as these are really ugly hacks. What do you think about running some kind of cron job that update the version in the package automatically?

**Comment by freswa** [comment]
The automatic update is pretty simple to implement. The main reason why this has not been implemented yet is the necessary (push) access to the AUR. While automatic builds are my daily business, I can't put my access to the AUR on a autonomous machine. That won't happen.

**Comment by mohe2015** [comment]
Another idea I had would be to put the $(date ...) into the pkgver but this is probably also considered hacky especially because the nightly packages are not always available at 0:00

Edit: this would also create lots of downloaded files as you would need to add the version to the source file to force redownloads.

Probably building from source using a git package would be the only sane option.

**Comment by freswa** [comment]
Why don't you just create a simple cron job on your machine? The PKGBUILD already fetches the latest nighly build. Manual intervention is only needed when there is a major version bump.

**Comment by mohe2015** [comment]
> Why don't you just create a simple cron job on your machine? The PKGBUILD already fetches the latest nighly build. Manual intervention is only needed when there is a major version bump.

I'm using an aur helper which doesn't support this kind of automatic update. But maybe I should just do it by hand. But thanks a lot for the ideas and nice responses.

**Comment by freswa** [comment]
If you need any help with your setup, feel free to drop me a message via e-mail or irc.

**Comment by mohe2015** [comment]
For now I decided to use the manual nightly download from the official website as that one at least autoupdates and is the least effort for me for now. But thanks for your help, it is quite rare to find such a nice person.

**Pull request closed without merging**